build-caching/src/main/java/com/vertispan/j2cl/build/DiskCache.java

```diff
@@ -436,9 +469,14 @@ private WatchKey registerWatchCreate(Path taskDir) throws IOException {
         return watchable.register(this.service, StandardWatchEventKinds.ENTRY_CREATE);
     }
 
-    public void markFinished(CacheResult successfulResult) {
+    public void markFinished(Input input, CacheResult successfulResult) {
         try {
-            this.knownOutputs.put(successfulResult.taskDir, makeOutput(successfulResult.taskDir));
+            TaskOutput output = makeOutput(successfulResult.taskDir);
+            this.lastSuccessfulTaskDir.put(input, successfulResult.taskDir);
+            this.knownOutputs.put(successfulResult.taskDir, output);
+            if (input.getOutputType().equals(OutputTypes.TRANSPILED_JS)) {
```

Comment:
I hard coded this here, as a hack. imho the writeFilesDat should be in it's own leaf task, only called once all other tasks are done.